chart/compass/charts/instance-creator/templates/service.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Chart.Name }}
+    release: {{ .Release.Name }}
+spec:
+  type: ClusterIP
+  ports:
+    - port: {{ .Values.global.instanceCreator.port }}
+      protocol: TCP
+      name: http
+    - port: {{ .Values.global.istio.proxy.port }}
+      protocol: TCP
+      name: proxy-status
+  selector:
```

Comment:
You can consider extracting these labels in a `_helpers.tpl` once again. Afterwards you can use the new helpers definition here, in the `.spec.matchLabels` and `.spec.template.metadata.labels` of the Deployment. This way you avoid issues if you change the labels, but might be an overkill if they are not going to change as often.

Reply:
I lean towards the latter because I don't think there will be many (if any) changes on a regular basis.